lib/core/test/unit/Cardano/Wallet/Primitive/CoinSelection/MA/RoundRobinSpec.hs

```diff
@@ -533,33 +579,52 @@ prop_performSelection minCoinValueFor costFor (Blind criteria) coverage =
             , pretty (Flat balanceSelected)
             , "change balance:"
             , pretty (Flat balanceChange)
+            , "absolute minimum coin quantity:"
+            , pretty absoluteMinCoinValue
+            , "actual coin delta:"
+            , pretty (TokenBundle.getCoin delta)
+            , "number of outputs:"
+            , pretty (length outputsCovered)
             ]
-        let delta = TokenBundle.unsafeSubtract
-                balanceSelected
-                (balanceRequired <> balanceChange)
+        assert True
         assert $ balanceSufficient criteria
         assert $ on (==) (view #tokens)
             balanceSelected (balanceRequired <> balanceChange)
-        assert $ TokenBundle.getCoin delta == expectedCost
+        assert $ TokenBundle.getCoin delta >= expectedCost
+        assert $ TokenBundle.getCoin delta <= maximumExpectedDelta
         assert $ utxoAvailable
             == UTxOIndex.insertMany inputsSelected utxoRemaining
         assert $ utxoRemaining
             == UTxOIndex.deleteMany (fst <$> inputsSelected) utxoAvailable
-        assert $ view #outputsCovered result == NE.toList outputsToCover
+        assert $ outputsCovered == NE.toList outputsToCover
         case selectionLimit of
             MaximumInputLimit limit ->
                 assert $ NE.length inputsSelected <= limit
             NoLimit ->
                 assert True
       where
+        absoluteMinCoinValue = mkMinCoinValueFor minCoinValueFor TokenMap.empty
+        delta = TokenBundle.unsafeSubtract
+            balanceSelected
+            (balanceRequired <> balanceChange)
+        maximumExpectedDelta =
+            expectedCost `addCoin`
+            (absoluteMinCoinValue `multiplyCoin`
+                (length outputsCovered - length changeGenerated))
```

Comment:
I'd expect the maximum delta to actually be: `expectedCost + absoluteMinCoinValue`. Did QC find counter-examples of that already :thinking: ?

Reply:
After checking and fiddling around with that, there are indeed cases where may not be able to create _any_ change output, in which case this property is indeed verified.